src/swim/swim.c

```diff
@@ -41,6 +41,40 @@
 #include "swim_internal.h"
 #include <assert.h>
 
+static uint64_t swim_ping_timeout = SWIM_PING_TIMEOUT;
+
+static inline void
+swim_dump_updates(swim_id_t self_id, swim_id_t from, swim_id_t to,
+		  struct swim_member_update *upds, size_t nupds)
+{
+	FILE *fp;
+	char *msg;
+	size_t msg_size, i;
+	int rc;
+
+	if (!D_LOG_ENABLED(DLOG_INFO))
+		return;
+
+	fp = open_memstream(&msg, &msg_size);
+	if (fp != NULL) {
+		for (i = 0; i < nupds; i++) {
+			rc = fprintf(fp, " {%lu %c %lu}", upds[i].smu_id,
+				     "ASD"[upds[i].smu_state.sms_status],
+				     upds[i].smu_state.sms_incarnation);
+			if (rc < 0)
+				break;
+		}
+
+		fclose(fp);
+		/* msg and msg_size will be set after fclose(fp) only */
+		if (msg_size > 0)
+			SWIM_INFO("%lu %s %lu:%s\n", self_id,
```

Comment:
FYI, this can also be cleaned up

This is defined to D_DEBUG(DLOG_INFO, fmt....)

It could just be #define SWIM_INFO D_INFO

Or you could also just use D_INFO directly though the above would be simpler.

Reply:
Ohhh. Probably it would be separate patch for this.